First try to manage PLATFORM GCC

straccio opened the pull request:
Now the GCC platform can connect correctly to the cloud

straccio committed:
First try to manage PLATFORM GCC

jlkalberer commented on src/lib/FirmwareManager.js:
```diff
@@ -39,6 +39,12 @@ class FirmwareManager {
     const platformID = systemInformation.p;
+
+    // GCC Platform skip OTA System
+    if(platformID==3){
```
nit: `platformID === 3`

straccio replied:
GCC have platform id = 3.
GCC doesn't have app module ad the app_hash

jlkalberer commented on src/server/DeviceServer.js:
```diff
@@ -361,9 +362,16 @@ class DeviceServer {
+      try {
+        uuid = FirmwareManager.getAppModule(
+          description.systemInformation,
+        )[0];
```
This doesn't seem correct. `find` should return  a single module.

straccio replied:
at this moment i can't try it with one real Photon.
`const {uuid} uuid = FirmwareManager.getAppModule(
          description.systemInformation,
);` 
is egual to       
`uuid = FirmwareManager.getAppModule(
          description.systemInformation,
)[0];`
I'm wrong?

jlkalberer replied:
Yeah -- `FirmwareManager.getAppModule` uses `find` which will return a single module.

The module returns `{uuid: string, .....otherStuff}`

when you do `const {uuid} = FirmWareManager.getAppModule` it uses destructuring so you only get the `uuid` prop out of the returned object.

I think the `platformID` check (above) will be fine but we probably don't need this change.

straccio replied:
Without GCC platform can't connect because FirmwareManager.getAppModule throw a null exception

jlkalberer replied:
The code you commented out should be here -- not the way you are using `getAppModule` now. It doesn't return an array and the code we had before was correct. 

This would solve your problem
```
let applicationID = 'none';
const appModule = FirmwareManager.getAppModule(
  description.systemInformation,
);
if (appModule) {
  applicationID = appModule.uuid;
}
```

jlkalberer reviewed (commented):
Can you explain what these checks are actually doing?

straccio committed:
Changed to binary operator

straccio committed:
Merge remote-tracking branch 'upstream/dev'

jlkalberer commented on src/server/DeviceServer.js:
```diff
@@ -347,6 +347,7 @@ class DeviceServer {
   _onDeviceReady = async (device: Device): Promise<void> => {
     try {
       logger.log('Device online!');
+      let uuid;
```
This definition should be closer to where it's set.
I think it should also be initialized to `null`

jlkalberer commented on src/server/DeviceServer.js:
```diff
@@ -361,9 +362,16 @@ class DeviceServer {
+        uuid = FirmwareManager.getAppModule(
+          description.systemInformation,
+        )[0];
+      } catch (uuidErr) {
```
nit: `} catch (ignore) {`

straccio committed:
Merge remote-tracking branch 'upstream/dev'

straccio committed:
Merge remote-tracking branch 'upstream/dev'

straccio committed:
Manage MONOLITHIC Firmware.

In Particle can exist Modular firmwares that return modules and Monolithic firmwares that's don't have modules and getModules returns null.

jlkalberer commented on src/lib/FirmwareManager.js:
```diff
@@ -38,6 +38,14 @@ class FirmwareManager {
     const platformID = systemInformation.p;
+
+    // GCC Platform skip OTA System
+    // if(platformID === 3){
```
Remove this. We don't allow commented code

jlkalberer commented on src/server/DeviceServer.js:
```diff
@@ -403,6 +410,70 @@ class DeviceServer {
     );
   };
 
+  // _onDeviceReady = async (device: Device): Promise<void> => {
```
???? What

straccio committed:
Removed comment code

jlkalberer reviewed (approved):
This looks ok.

You only changed the code in two places right? The rest was from prettier?

straccio committed:
Merge remote-tracking branch 'upstream/dev'

straccio committed:
Merge

straccio committed:
Merge remote-tracking branch 'upstream/dev'

jlkalberer commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "spark-protocol",
   "version": "0.1.6",
-  "main": "./dist/index.js",
+  "main": "./src/index.js",
```
nit: revert

jlkalberer commented on src/defaultBindings.js:
```diff
@@ -61,6 +61,11 @@ const defaultBindings = (
     ['DEVICE_DIRECTORY'],
   );
+  // <<<<<<< HEAD
+  //   container.bindClass('DeviceKeyRepository', DeviceKeyFileRepository, [
```
Remove this

jlkalberer commented on src/defaultBindings.js:
```diff
@@ -69,6 +74,7 @@ const defaultBindings = (
     'IProductFirmwareRepository',
     MockProductFirmwareRepository,
   );
+  // >>>>>>> upstream/dev
```
remove

jlkalberer commented on src/defaultBindings.js:
```diff
@@ -79,16 +85,24 @@ const defaultBindings = (
   container.bindClass('ClaimCodeManager', ClaimCodeManager, []);
   container.bindClass('CryptoManager', CryptoManager, [
+    // <<<<<<< HEAD
+    //     'DeviceKeyRepository',
```
Remove

jlkalberer commented on src/defaultBindings.js:
```diff
@@ -79,16 +85,24 @@ const defaultBindings = (
   // Device server
   container.bindClass('DeviceServer', DeviceServer, [
+    // <<<<<<< HEAD
+    //     'DeviceAttributeRepository',
```
Remove

jlkalberer commented on src/server/DeviceServer.js:
```diff
@@ -367,11 +367,23 @@ class DeviceServer {
+
+          const { uuid: appHash } = appModules;
+          //= ======
+          //           const appModule = FirmwareManager.getAppModule(systemInformation);
```
Remove

jlkalberer reviewed (changes requested):
I'm not sure why you're touching all these files when it's mostly a 3 line addition.  Is this `prettier` doing its thing?

Also, please remove all the merge conflicts. This should not be merged into master.

straccio committed:
Corrections
